sql/core/src/test/scala/org/apache/spark/sql/execution/command/CreatePipelineDatasetAsSelectParserSuiteBase.scala

```diff
@@ -0,0 +1,239 @@
+              defaultValue = None,
+              generationExpression = None,
+              identityColumnSpec = None,
+              metadata = new MetadataBuilder().build()
+            ),
+            ColumnDefinition(
+              name = "b",
+              dataType = StructType(
+                Seq(
+                  StructField("field", StringType, true),
+                  StructField("field2", IntegerType, true)
+                )
+              ),
+              nullable = true,
+              comment = Some("column description comment"),
+              defaultValue = None,
+              generationExpression = None,
+              identityColumnSpec = None,
+              metadata = new MetadataBuilder().build()
+            )
+          )
+      ),
+      (s"CREATE $datasetSqlSyntax table1 AS SELECT * FROM input", Seq.empty)
+    ).foreach { case (query, columnDefs) =>
+      val plan = parser.parsePlan(query)
+      val cmd = plan.asInstanceOf[CreatePipelineDatasetAsSelect]
+      assert(cmd.columns == columnDefs)
+    }
+  }
+
+  test("Partitioning is correctly parsed") {
+    Seq(
+      (s"CREATE $datasetSqlSyntax table1 PARTITIONED BY (a) AS SELECT * FROM input",
+        Seq(IdentityTransform(FieldReference(Seq("a"))))),
+      (s"CREATE $datasetSqlSyntax table1 AS SELECT * FROM input", Seq.empty)
+    ).foreach { case (query, partitioning) =>
+      val plan = parser.parsePlan(query)
+      val cmd = plan.asInstanceOf[CreatePipelineDatasetAsSelect]
+      assert(cmd.partitioning == partitioning)
+    }
+  }
+
+  test("Location is unsupported") {
+    val ex = intercept[ParseException] {
+      parser.parsePlan(
+        s"""CREATE $datasetSqlSyntax table1 LOCATION 'dbfs:/path/to/table'
+           |AS SELECT * FROM input""".stripMargin)
+    }
+    assert(ex.getMessage.contains("Specifying location is not supported for " +
+      s"CREATE $datasetSqlSyntax statements."))
+  }
+
+  test("Column constraints are unsupported") {
+    val ex = intercept[ParseException] {
+      parser.parsePlan(s"CREATE $datasetSqlSyntax table1 (id INT CHECK(id > 0)) AS SELECT 1")
+    }
+    assert(ex.getMessage.contains("Pipeline datasets do not currently support column constraints"))
+  }
+
+  test("Comment is correctly parsed") {
+    Seq(
+      (s"CREATE $datasetSqlSyntax table1 COMMENT 'this is a comment' AS SELECT * FROM input",
+        Some("this is a comment")),
+      (s"CREATE $datasetSqlSyntax table1 AS SELECT * FROM input", None)
+    ).foreach { case (query, commentOpt) =>
+      val plan = parser.parsePlan(query)
+      val cmd = plan.asInstanceOf[CreatePipelineDatasetAsSelect]
+      assert(cmd.tableSpec.comment == commentOpt)
+    }
+  }
+
+  test("STORED AS syntax is unsupported") {
+    val ex = intercept[ParseException] {
+      parser.parsePlan(s"CREATE $datasetSqlSyntax table1 STORED AS TEXTFILE AS SELECT * FROM input")
+    }
+    assert(ex.getMessage.contains("The STORED AS syntax is not supported for CREATE " +
```

Comment:
For the rest of the test cases: let's use `checkError` instead of `assert(ex.getMessage.contains...`